flaml/automl.py

```diff
@@ -507,7 +509,8 @@ def custom_metric(
                 True - retrain only after search finishes; False - no retraining;
                 'budget' - do best effort to retrain without violating the time
                 budget.
-            split_type: str, default="auto" | the data split type.
+            split_type: str or iterable, object that has split and get_n_splits methods,
```

Comment:
what is the use case of having split_type in iterable type? Maybe elaborate a bit or give an example.

Reply:
Please add a test in test/automl/test_split.py for both cases: iterable, and class with split/get_n_splits methods.

Reply:
We may need to elaborate what are the constraints for `split_type` when it is iterable, or object -- what is input/output for `split()` and `get_n_split()`.

Do you think it would be easier if we have one single API for customized splitter: users implement a derived class of https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold, customizing split() and get_n_splits() function but with the same signature. `Iterable split_type` can be implemented using this API as well. This way, we have a more concise API and do not need to worry about whether the input `iterable split_type` is in correct form or not.

Reply:
> We may need to elaborate what are the constraints for `split_type` when it is iterable, or object -- what is input/output for `split()` and `get_n_split()`.
> 
> Do you think it would be easier if we have one single API for customized splitter: users implement a derived class of https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold, customizing split() and get_n_splits() function but with the same signature. `Iterable split_type` can be implemented using this API as well. This way, we have a more concise API and do not need to worry about whether the input `iterable split_type` is in correct form or not.

Yeah, the definitions of an `iterable`, `split()` and `get_n_split()` are not very clear. It is clearer to use a derived class of `KFold`. I can try to change it to that. Do I need to open a new PR then.

Reply:
Is it hard to check the signature during runtime? If so, maybe we can just write down the API signature of `split()` and `get_n_split()` in example/comments/docs and catch the exception during runtime and report the error?

Reply:
OK, I can try this